components/prism-graphql.js

```diff
@@ -18,7 +18,16 @@ Prism.languages.graphql = {
 	},
 	'number': /(?:\B-|\b)\d+(?:\.\d+)?(?:e[+-]?\d+)?\b/i,
 	'boolean': /\b(?:true|false)\b/,
+	'variable-input': {
+		pattern: /(\bmutation.*)\$\w+/,
+		alias: 'variable',
+		lookbehind: true
+	},
 	'variable': /\$[a-z_]\w*/i,
+	'atom-input': {
+		pattern: /[A-Z]\w*Input(?=!?.*$)/m,
```

Comment:
Managed to simplify this pattern a bit to get rid of the `mutation.*` [and fix a test that slipped by because I missed the diff until I looked at the snapshot 😅 ] but yeah I'm not sure if I can refactor the `variable-input` and `property-mutation` tokens like this. 

If someone can piece together a better pattern from those tests I would be very impressed.

Reply:
@RunDevelopment I found in my testing that we do need this one to get the input type definitions correct... I'll add it back with a test.

Reply:
Could you share the example that was wrong?

Reply:
Here's the failing example:

```
input OrderInput {
	id: ID!
	customAttributes: [AttributeInput!]
	customer: Customer
	localizationExtensions: [LocalizationExtensionInput!]
	metafields: [MetafieldInput!]
	note: String!
	shippingAddress: MailingAddressInput
	tags: [String!]
}
```

and I added a failing test for it in my [`tests/e2e-example` branch](https://github.com/JaKXz/prism/pull/1/files#diff-bd57f122459de3b3ce8bc9288301d18ae02ce2ceac7e4937a4039c3cd7f8bb24R26) which passes if I put this pattern back in.

Reply:
This is fixed in my branch https://github.com/JaKXz/prism/pull/1 -- I won't merge that while we're not on board with the move to snapshot e2e testing :) thanks for the discussion and all the help though!